thriftbp/thrifttest/server.go

```diff
@@ -131,20 +141,34 @@ func (s *Server) Close() error {
 // This is inspired by httptest.NewServer from the go standard library and can
 // be used to test a thrift service.
 func NewBaseplateServer(cfg ServerConfig) (*Server, error) {
-	socket, err := thrift.NewTServerSocket(loopbackAddr)
-	if err != nil {
-		return nil, err
-	}
-	// Call listen to reserve a port and check for any issues early
-	if err := socket.Listen(); err != nil {
-		return nil, err
+	var socket *thrift.TServerSocket
+	var addr string
+	if cfg.TB != nil {
+		dir := cfg.TB.TempDir()
+		path := filepath.Join(dir, "socket")
+		socket = thrift.NewTServerSocketFromAddrTimeout(&net.UnixAddr{
+			Net:  "unix",
+			Name: path,
+		}, 0)
+		addr = "uds://" + path
```

Comment:
I think this code will likely fail if ran on windows, but I don't think we care about windows users 🤷 

if we do care about them, then we can extract this into 2 files with build conditions.